mesonbuild/interpreter.py

```diff
@@ -2177,19 +2200,20 @@ def dependency_fallback(self, name, kwargs):
         if not isinstance(dep, DependencyHolder):
             raise InvalidCode('Fallback variable {!r} in the subproject {!r} is '
                               'not a dependency object.'.format(varname, dirname))
+
         # Check if the version of the declared dependency matches what we want
-        if 'version' in kwargs:
-            wanted = kwargs['version']
-            found = dep.version_method([], {})
-            if found == 'undefined' or not mesonlib.version_compare(found, wanted):
-                mlog.log('Subproject', mlog.bold(dirname), 'dependency',
-                         mlog.bold(varname), 'version is', mlog.bold(found),
-                         'but', mlog.bold(wanted), 'is required.')
-                return None
-        mlog.log('Found a', mlog.green('fallback'), 'subproject',
-                 mlog.bold(os.path.join(self.subproject_dir, dirname)), 'for',
-                 mlog.bold(name))
-        return dep
+        version = kwargs.get('version')
+        dep = self.check_version (name, dep, version, dirname, varname)
+        if dep:
+            found_msg = ['Fallback subproject dependency',
+                mlog.bold(name), 'found:', mlog.green('YES'),
+                '({})'.format(os.path.join(self.subproject_dir, dirname))]
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->